components/scale/dateTime.js

```diff
@@ -60,6 +60,35 @@
             return scale;
         };
 
+        scale.nice = function() {
+            adaptedScale.nice();
+            var domain = adaptedScale.domain();
+            var domainLower = discontinuities().clampUp(domain[0]);
+            var domainUpper = discontinuities().clampDown(domain[1]);
+            adaptedScale.domain([domainLower, domainUpper]);
+            return scale;
+        };
+
+        scale.ticks = function() {
+            var ticks = adaptedScale.ticks();
+            var clampedTicks = ticks.map(function(tick, index) {
+                if (index < ticks.length - 1) {
```

Comment:
No, you have to clamp the sides inwards, otherwise you risk clamping  the upper or lower ticks outside of the axis range.